### ISSUE TITLE
JavaScript browser console error: showFailuresLink is null

### ISSUE THREAD

**Issue by jayaddison** [jenkinsci/junit-plugin#513]
### Jenkins and plugins versions report

Jenkins: 2.387.1

### What Operating System are you using (both controller, and any agents involved in the problem)?

OS: Debian Linux (bookworm)
Browser: Firefox ESR (102.9)

### Reproduction steps

1. Visit https://jenkins.sqlalchemy.org/job/openstack_gerrit/11736/
2. Open Firefox "Web Developer Tools" (ctrl+shift+I)
3. Observe JavaScript error in the console: `Uncaught TypeError: showFailuresLink is null`

### Expected Results

No JavaScript errors should occur when displaying the results of a Jenkins build job.

### Actual Results

A JavaScript error, `Uncaught TypeError: showFailuresLink is null` appears in the developer console.

### Anything else?

Some code relating to this was [updated in 2022](https://github.com/jenkinsci/junit-plugin/commit/1c24f1ad2553ef40f1801edc752c654572664066) - those changes are largely a relocation of logic from one place to another.

It's possible that there could be a subtle issue related to an [`if` condition](https://github.com/jenkinsci/junit-plugin/commit/1c24f1ad2553ef40f1801edc752c654572664066#diff-62f4d594cb5dd6eb6dd61a230ac35877af64dae4fb19e54fa3e7167298404147L76) in the server-side content rendering.

In particular:

- The `showLink` element was never guaranteed be placed on the page by this plugin
- Although the code before 1c24f1ad2553ef40f1801edc752c654572664066 did include a `document.getElementById("showLink")` call (potentially evaluating to `null`), that was only callable when the `showLink` element itself was displayed on the page
- After the changes, it seems like the relevant query may be called during `DOMContentLoaded` event handling

**Comment by timja** [comment]
cc @aneveux  if you're interested.

@jayaddison are you interested in contributing a fix for this?

**Comment by jayaddison** [comment]
@timja I'd like to try, yep - I'm not familiar with the developer tooling for this project, but most of the fix should be plain JavaScript, I think (aside from the process of testing it).

Two options spring to mind:

- Move the `showFailures` event handler back to the `showLink` HTML element, and remove the `DOMContentLoaded` event handler

OR

- Conditionally check for a non-null `showFailuresLink` object before binding an event handler in the `show-failures.js` script

The first option would be more like a partial-revert, the second option would maintain more separation between the HTML and JS (although.. it would make it less obvious -- no console errors -- if that code gets included on a page where it shouldn't be).

**Comment by timja** [comment]
Second option please.
It should check for existence of the element before acting on it if it might not be on the page.

you can start up a dev server with `mvn hpi:run` where you can test it interactively

**Comment by uhafner** [comment]
BTW: duplicate of #480

**Comment by jayaddison** [comment]
I now think that this console error occurs either when there are no test failures from a build, or when the number of failures is less-than-or-equal to ten.

However: I'm having trouble reproducing the problem locally.  I have created and (re)configured a job to produce both a successful build and also a failed build, but the browser console error is not appearing.

Neither is the [clipboard icon](https://github.com/jenkinsci/junit-plugin/blob/19a5192cc1fc758a680695463392c2388318f59a/src/main/resources/hudson/tasks/test/AbstractTestResultAction/summary.jelly#L32) that I'd expect to find in the rendered HTML.  So perhaps the relevant `jelly` module isn't being included for some reason.

**Comment by jayaddison** [comment]
> However: I'm having trouble reproducing the problem locally. I have created and (re)configured a job to produce both a successful build and also a failed build, but the browser console error is not appearing.

The problem here was that I hadn't configured a valid `junit` step as part of the build.  Adding that (by producing a simple JUnit report using `writeFile`, and then processing it with a `junit` step) provided repro of the problem.

The README helped to figure this out, and requires a small update that I'll provide soon.

@timja about this:
> Second option please.
> It should check for existence of the element before acting on it if it might not be on the page.

Although I began developing that, after starting, a third option appeared more sensible: we can move the inclusion of the relevant `show-failures.js` script into the same conditional block where the `showLink` element is rendered.  I've opened #514 with that change.
